### DISCUSSION TITLE
[FeatureRequest] enable the use of storybook-react for tamagui

### DISCUSSION THREAD

**Discussion by dohomi** [tamagui/tamagui#232] [General]
**Is your feature request related to a problem? Please describe.**
Enable the use of tamagui components with Storybook.

**Describe the solution you'd like**
I tried to set up tamagui with storybook-react but the context of the theme is getting lost. It works with storybook-react-native but not with storybook-react. All components which require the theme are throwing an error that the config could not get loaded. 

Clone https://github.com/dohomi/tamagui-kitchen-sink 
Run `yarn && yarn sb`. All components where a Button is present is throwing an error.

Initially this repo tried it in a similar way https://github.com/aleedmanath/tamagui-storybook-test but fails with the same error.

**Comment by dohomi** [comment]
Ok I find digged a bit deeper and found a solution to run storybook-react with https://github.com/dohomi/tamagui-kitchen-sink. storybook-vite is not working yet but I will have a look at it another time

**Reply by pfcodes** [reply]
What was the solution?

**Reply by dohomi** [reply]
Just have a look at the repo where you can find a working solution for Storybook. There are also couple of issues open on Discord

**Comment by pontusab** [comment]
this is great! 💯

**Comment by pontusab** [comment]
Noticed that in build mode it will fail because missing `process.NODE_DEBUG`, I added this to the `main.ts` config file:

```
async viteFinal(config) {
    return mergeConfig(config, {
      plugins: [
        tamaguiPlugin({
          ...tamaguiConfig,
          useReactNativeWebLite: false,
        }),
        tamaguiExtractPlugin(tamaguiConfig),
      ],
      define: {
        'process.env.NODE_DEBUG': false, <-- this
      },
    });
  },
```

**Comment by dohomi** [comment]
@pontusab do you mind create a PR at the kitchen sink? I did several tryouts but never succeeded with storybook-vite

**Comment by pontusab** [comment]
@dohomi So I got everything working locally, running storybook-build, and then started the webserver to try the storybook out, everything works. But when I added everything to the workflow in GitHub actions (runs the same command it then fails) missing index.html, I tracked it down to when I'm adding the Tamagui plugin `tamaguiPlugin`, I don't really know why this happens, I will create a repo with a workflow for us. I went with the webpack solution for now to unblock us.

**Comment by dohomi** [comment]
Ok, the webpack version works fine for me as well I am starting to add more and more usecases into the ui to see how to address more usecases.

**Comment by dohomi** [comment]
@natew unfortunately after tamagui beta.185 the webpack version of Storybook stops compiling. It stops at 57% and thats it. I tried to debug and tried all other version between 185 > 191 but none of them work with Storybook any more. I had to pin the tamagui version to `184`. It would be great if you could have a look what might have broke it.

**Reply by dohomi** [reply]
@pontusab did you experience the same in your setup?

**Comment by dohomi** [comment]
Ok I found some very strange issue: I have `@tamagui/config-base` as a dependency in my `packages/config` folder -  which is actually unused (was a tryout in the past). If I remove that, then `Storybook` stops building. If I add it, then it works until version `185`. This seems like a babel/webpack conflict?

**Comment by kibiz0r** [comment]
@dohomi I'm also running into the "Missing tamagui config" error, and I've tried mimicking your setup in a trimmed-down, non-monorepo context, with no luck. I'm not sure what I'm missing.

Please, could you give a bit more detail about how you worked around the config error? What was the root cause?

For reference, you can see what I've got here: https://github.com/kibiz0r/tamagui-storybook-config-error

**Comment by dohomi** [comment]
your example is covering `@storybook/react-native` while my example is using `@storybook/react`. I haven't got the time to dig into the native part yet but consider checking out the `Takeout` which I think delivers a working Storybook configuration https://tamagui.dev/takeout

**Comment by kibiz0r** [comment]
@dohomi Thank you for your reply! I'm just now getting a chance to come back to try this again.

Looks like I had some old `.ondevice` files in there that mentioned `@storybook/react-native`, but that package wasn't even installed so those files would've just blown up if anything brought them in. I just pushed an update removing them, with no change in the "missing config" error.

So it looks like it really has only been using `@storybook/react` and `react-native-web`, very much like the storybook-react part of your kitchen-sink setup.

If you would please provide some info about what you ran into and what you identified as the root cause or the critical part of the solution, it would really mean a lot to me as it would help me direct my efforts.

**Comment by dohomi** [comment]
I honestly dont have any more advise than looking into the Github repo and see how I've done the config part. Since Storybook v7 and the newest iteration on Tamagui itself the integration was more less seemlessly. Also you can consider to purchase the Tamagui Takeout, as that has even more bulletproof way how to integrate Tamagui with all NextJS, Vite, Storybook and some more.

**Comment by quyenhuynh-agilityio** [comment]
I got an error `Error: Haven't called createTamagui yet at getTokens`.
Do you have any updates?
